src/components/CommunicationsLink.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import {View, Pressable, Linking} from 'react-native';
+import PropTypes from 'prop-types';
+import styles from '../styles/styles';
+import compose from '../libs/compose';
+import {Checkmark, Clipboard as ClipboardIcon} from './Icon/Expensicons';
+import Clipboard from '../libs/Clipboard';
+import ContextMenuItem from './ContextMenuItem';
+import withLocalize, {withLocalizePropTypes} from './withLocalize';
+import withWindowDimensions, {windowDimensionsPropTypes} from './withWindowDimensions';
+import CONST from '../CONST';
+
+const propTypes = {
+    /** Children to wrap in CommunicationsLink. */
+    children: PropTypes.node.isRequired,
+
+    /** Styles to be assigned to Container */
+    containerStyles: PropTypes.arrayOf(PropTypes.object),
+
+    /** Decides Tap behaviour. */
+    type: PropTypes.oneOf([CONST.LOGIN_TYPE.PHONE, CONST.LOGIN_TYPE.EMAIL]),
+
+    /** Value to be copied or passed via tap. */
+    value: PropTypes.string.isRequired,
+
+    ...windowDimensionsPropTypes,
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    containerStyles: [],
+    type: undefined,
+};
+
+const CommunicationsLink = props => (
+    <View style={[styles.flexRow, styles.pRelative, ...props.containerStyles]}>
+        {props.type && props.isSmallScreenWidth
+            ? (
+                <Pressable
+                    onPress={() => Linking.openURL(
+                        props.type === CONST.LOGIN_TYPE.PHONE
+                            ? `tel:${props.value}`
+                            : `mailto:${props.value}`,
+                    )}
+                >
+                    {props.children}
+                </Pressable>
+            )
+            : props.children}
+        {props.type && !props.isSmallScreenWidth
+            && (
+                <View style={[
+                    styles.pAbsolute,
+                    styles.alignItemsCenter,
+                    styles.justifyContentCenter,
+                    styles.communicationsLinkIcon]}
+                >
+                    <ContextMenuItem
+                        icon={ClipboardIcon}
+                        text={props.translate('contextMenuItem.copyToClipboard')}
+                        successIcon={Checkmark}
+                        successText={props.translate('contextMenuItem.copied')}
+                        isMini
+                        autoReset
+                        onPress={() => Clipboard.setString(props.value)}
+                    />
+                </View>
+            )}
```

Comment:
Still don't get the checks for `props.type`. What is using this without a type? If there's no `type` we show `props.children`? We should not use it at all in that case.

Reply:
To make `CommunicationLink` uses cleaner I did this. Otherwise, we have to conditionally show the content wrapped in `CommunicationLink` which will duplicate a few lines on the main Component and This approach gives it a cleaner usage. https://github.com/parasharrajat/Expensify.cash/blob/a6814422a60fbba13d04e19f2807190a82450c2f/src/pages/DetailsPage.js#L95.

I will welcome suggestions to improve it.

Reply:
Ok, I am removing the confusing uses of Type. Instead, just use the conditional block here https://github.com/parasharrajat/Expensify.cash/blob/a6814422a60fbba13d04e19f2807190a82450c2f/src/pages/DetailsPage.js#L95

Reply:
Perfect thanks! 

> Otherwise, we have to conditionally show the content wrapped in CommunicationLink which will duplicate a few lines on the main Component and This approach gives it a cleaner usage.

I have a different opinion. To me it's not clear why we should use a component that _might_ do nothing at all. To me, it's as if we are calling a function with an optional argument that specifies whether we should call the function or not. In this case, it's clearer to just not call the function in the first place.

Reply:
Thanks for clarifying.